### PR TITLE
fix(cli): ignore current review run in presubmit CI

### LINKED ISSUE #6396: Qwen /review can downgrade approval while waiting on its own review check
## What happened?

The automated Qwen Code `/review` flow can downgrade an otherwise approving review to a plain comment because it counts the currently running review workflow's own check-run as pending CI.

In PR #6387, `qwen-code-ci-bot` found no issues and produced an LGTM verdict, but the published review state was `COMMENTED` with:

```text
Downgraded from Approve to Comment: CI still running.
```

The review was submitted at `2026-07-06T12:29:02Z`, while the same `🧐 Qwen Pull Request Review / review-pr` job for that run did not finish until `2026-07-06T12:30:11Z`. Because `/review presubmit` checks all check-runs attached to the PR HEAD commit, it saw its own `review-pr` check-run as `in_progress` at the moment it submitted the review.

That makes the approval gate self-referential: the review job cannot approve while the review job itself is still running.

## What did you expect to happen?

The current Qwen PR review workflow run should not block its own approval decision. Real PR CI and commit statuses should still block approval when they are pending or failing, but the current review workflow's own check-run should be ignored for this specific presubmit CI classification.

## Client information

<details>
<summary>Client Information</summary>

Maintainer-side GitHub Actions `/review` workflow issue observed on QwenLM/qwen-code PR #6387.

</details>

## Login information

GitHub Actions workflow using `qwen-code-ci-bot`.

## Anything else we need to know?

Evidence from PR #6387:

- Review URL: https://github.com/QwenLM/qwen-code/pull/6387#pullrequestreview-4635826466
- Review submitted: `2026-07-06T12:29:02Z`
- `review-pr` job completed: `2026-07-06T12:30:11Z`
- The review body said it was downgraded because CI was still running, even though the pending check was the review workflow itself.

### PR DESCRIPTION
**Pull request by yiliang114**
## What this PR does

Updates `/review presubmit` CI classification so the currently running GitHub Actions review run does not count as pending CI for its own approval decision. The filter is intentionally narrow: it only ignores check-runs whose `details_url` or `html_url` belongs to the current `GITHUB_RUN_ID`.

Adds a regression test for the observed failure mode where the real PR CI has already completed successfully but `🧐 Qwen Pull Request Review / review-pr` is still `in_progress`.

## Why it's needed

The automated Qwen Code review flow can otherwise become self-referential. In PR #6387, the bot found no issues and produced an approving verdict, but the published review was downgraded to `COMMENTED` with `CI still running` because the `review-pr` job was still running at the moment it submitted the review.

This prevents a clean bot approval from clearing stale bot review state, even when the only pending check is the review workflow itself. Real PR CI and commit statuses still participate in the pending/failing gate.

such as  https://github.com/QwenLM/qwen-code/pull/6397#issue-4819339885

image

### How to verify

Run the focused review command tests and CLI typecheck:

```console
cd packages/cli
npm run test -- src/commands/review.test.ts src/commands/review/presubmit.test.ts src/commands/review/post-suggestions.test.ts src/commands/review/pr-context.test.ts
npm run typecheck
```

The new regression test should show that a pending check-run from the current `GITHUB_RUN_ID` is ignored, while the remaining successful PR CI lets the presubmit report stay `all_pass` without a `CI still running` downgrade reason.

### Evidence (Before & After)

N/A — non-UI CI classification fix.

### Tested on

|     OS     | Status |
| :--------: | :----: |
|  🍏 macOS  |   ✅   |
| 🪟 Windows |   ⚠️   |
|  🐧 Linux  |   ⚠️   |

### Environment (optional)

Local worktree under `/tmp/qwen-review-ci-self-check`; `npm ci` completed before verification.

## Risk & Scope

- Main risk or tradeoff: filtering too broadly could hide real CI. This PR avoids that by matching only the current Actions run URL via `GITHUB_RUN_ID`, not by job name or workflow name.
- Not validated / out of scope: changing local `/review` behavior when no `GITHUB_RUN_ID` is present. Local runs continue to classify remote pending checks the same way as before.
- Breaking changes / migration notes: none.

## Linked Issues

Resolves #6396

<details>
<summary>中文说明</summary>

## 这个 PR 做了什么

更新 `/review presubmit` 的 CI 分类逻辑，让当前正在运行的 GitHub Actions review run 不再把自己的 check-run 当作 pending CI。过滤范围刻意保持很窄：只忽略 `details_url` 或 `html_url` 属于当前 `GITHUB_RUN_ID` 的 check-run。

新增回归测试覆盖已观察到的问题：真实 PR CI 已经成功完成，但 `🧐 Qwen Pull Request Review / review-pr` 自己仍处于 `in_progress` 时，不应把审批降级成 comment。

## 为什么需要

自动 Qwen Code review 流程之前会形成自引用等待。PR #6387 中，bot 没发现问题并给出 approve verdict，但实际发布的 review 被降级成 `COMMENTED`，原因是 `CI still running`；当时正在 pending 的正是同一个 `review-pr` job 自己。

这会阻止干净的 bot approve 清掉旧的 bot review 状态，即使唯一 pending 的 check 是 review workflow 自己。真实 PR CI 和 commit status 仍然会继续参与 pending/failing gate。

### How to verify

运行聚焦的 review 命令测试和 CLI typecheck：

```console
cd packages/cli
npm run test -- src/commands/review.test.ts src/commands/review/presubmit.test.ts src/commands/review/post-suggestions.test.ts src/commands/review/pr-context.test.ts
npm run typecheck
```

新增回归测试应证明：来自当前 `GITHUB_RUN_ID` 的 pending check-run 会被忽略；其余真实 PR CI 成功时，presubmit report 保持 `all_pass`，并且不会出现 `CI still running` 降级原因。

### Evidence (Before & After)

N/A — 非 UI 的 CI 分类修复。

### Tested on

|     OS     | Status |
| :--------: | :----: |
|  🍏 macOS  |   ✅   |
| 🪟 Windows |   ⚠️   |
|  🐧 Linux  |   ⚠️   |

### Environment (optional)

本地 worktree：`/tmp/qwen-review-ci-self-check`；验证前已完成 `npm ci`。

## Risk & Scope

- 主要风险或取舍：如果过滤过宽，可能隐藏真实 CI。本 PR 只通过 `GITHUB_RUN_ID` 匹配当前 Actions run URL，不按 job name 或 workflow name 泛化过滤。
- 未验证 / 不在范围内：没有 `GITHUB_RUN_ID` 的本地 `/review` 行为。本地运行仍和之前一样分类远端 pending checks。
- Breaking changes / migration notes：无。

## Linked Issues

Resolves #6396

</details>

### CODE REVIEW

File: packages/cli/src/commands/review/presubmit.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    } else {
+      process.env['GITHUB_RUN_ID'] = originalGithubRunId;
+    }
+  });
+
+  it('ignores the running Qwen PR review check when deciding whether CI is still pending', async () => {
+    ghApiMock.mockImplementation((path: string) => {
+      if (path.endsWith('/check-runs')) {
+        return {
+          check_runs: [
```
Comment (doudouOUC) [inline packages/cli/src/commands/review/presubmit.test.ts:86]: **[Suggestion]** Test coverage is thin — only the happy path is tested.

Several code paths introduced in this diff lack coverage:
1. `GITHUB_RUN_ID` unset — the early-return `false` path in `isCurrentActionsRunCheck` is never exercised
2. `html_url`-only matching — the new interface field is never tested as the sole match source
3. `totalChecks` value after filtering is never asserted (e.g., `expect(result.ciStatus.totalChecks).toBe(1)`)
4. The `.not.toContain('CI still running')` assertion could false-pass if the filter breaks but produces empty `downgradeReasons` for a different reason — consider `expect(result.downgradeReasons).toEqual([])` instead
5. Edge case where the self-check is the only check run (currently classifies as `no_checks` — should this be tested/documented?)

Sibling test files like `post-suggestions.test.ts` have 7+ cases. Consider adding 3-4 more `it` blocks.

_— qwen3.7-max via Qwen Code /review_

File: packages/cli/src/commands/review/presubmit.ts [hunk 2]
```diff
@@ -74,8 +86,11 @@ interface PresubmitArgs {
 function classifyCi(checkRuns: CheckRun[], statuses: CommitStatus[]) {
   const failedCheckNames: string[] = [];
   let hasPending = false;
+  const relevantCheckRuns = checkRuns.filter(
```
Comment (doudouOUC) [inline packages/cli/src/commands/review/presubmit.ts:89]: **[Suggestion]** Silent filter with no observability in the output JSON.

`classifyCi` filters check runs silently — the output JSON (`class`, `failedCheckNames`, `totalChecks`) contains no field indicating how many checks were filtered or which ones were excluded. There is no logging in the filter path.

At 3 AM when debugging "why did the bot approve despite failing CI," an operator cannot determine from the presubmit report alone whether the filter excluded something it shouldn't have.

Consider adding a `filteredSelfChecks` field to the output listing excluded check-run names, or at minimum a `writeStderrLine` when the filter activates.

_— qwen3.7-max via Qwen Code /review_

File: packages/cli/src/commands/review/presubmit.ts [hunk 3]
```diff
@@ -106,7 +121,7 @@ function classifyCi(checkRuns: CheckRun[], statuses: CommitStatus[]) {
   return {
     class: cls,
     failedCheckNames,
-    totalChecks: checkRuns.length + statuses.length,
+    totalChecks: relevantCheckRuns.length + statuses.length,
```
Comment (doudouOUC) [inline packages/cli/src/commands/review/presubmit.ts:124]: **[Suggestion]** `totalChecks` is ambiguous after filtering.

`totalChecks: relevantCheckRuns.length + statuses.length` uses the post-filter count. When the self-check is the only check run, `totalChecks` reports 0 — indistinguishable from a repository with no CI configured at all.

Consider either keeping `totalChecks` as the pre-filter count and adding a separate `filteredChecks` count, or renaming the field to `relevantChecks` to make the post-filter semantics explicit.

_— qwen3.7-max via Qwen Code /review_

**Review by doudouOUC — COMMENTED** [review]
## Qwen Code Review Summary for PR #6397

**Review completed** with 3 inline comments posted (all suggestions, no blockers).

### Findings

1. **Test coverage** — `presubmit.test.ts` focuses on a single happy-path scenario; several new code paths in `presubmit.ts` (e.g., `GITHUB_RUN_ID` unset, non-matching URLs, mixed status states) are not exercised.
2. **Observability** — `classifyCi` filters out the current Actions run silently. The output JSON and logs expose no field (e.g., `filteredSelfRuns`) indicating what was excluded, making future misclassifications hard to debug.
3. **`totalChecks` ambiguity** — The reported `totalChecks` uses the post-filter count, so a PR whose only check is the review run itself reports `0` checks, indistinguishable from a repository with no CI.

### Verdict

No critical issues found. The fix is correct and narrowly scoped — URL substring matching with `/actions/runs/${runId}/` is safe, and the filter is applied consistently across all usage sites. The inline comments are non-blocking suggestions for additional coverage and observability.

**Review by wenshao — APPROVED** [review]

**Comment by yiliang114** [comment]
本轮已处理 review comments：

- 补充测试覆盖 `GITHUB_RUN_ID` unset 时不应过滤 check-run。
- 补充测试覆盖仅 `html_url` 匹配当前 Actions run 的过滤路径。
- 在现有 happy path 中断言过滤后的 `totalChecks`，并将 downgrade reason 断言收紧为 `[]`。
- 未新增 `filteredSelfChecks`、stderr warning 或调整 `totalChecks` 语义：这些会改变 presubmit 输出契约/观测性设计，超出本 PR 对自引用 CI 过滤的最小修复范围。
- 未单独固化 “self-check 是唯一 check run” 场景：当前行为仍为过滤后 `no_checks`，是否改变或文档化属于输出语义决策，留给后续。

验证：`npm run test -- src/commands/review.test.ts src/commands/review/presubmit.test.ts src/commands/review/post-suggestions.test.ts src/commands/review/pr-context.test.ts`、`npm run typecheck`（在 `packages/cli`）。
